pennylane/transforms/__init__.py

```diff
@@ -265,11 +266,36 @@ def circuit(x, y):
 In this example, inverses are canceled, leading to the removal of two Hadamard gates. Subsequently, rotations are
 merged into a single :class:`qml.Rot` gate. Consequently, two transforms are successfully applied to the circuit.
 
+
+Passing arguments to transforms
+-------------------------------
+
+We can decorate a QNode with ``@partial(transform_fn, **transform_kwargs)`` to provide additional keyword arguments to a transform function.
+In the following example, we pass the keyword argument ``grouping_strategy="wires"`` to the :func:`~.split_non_commuting` quantum transform.
+
+.. code-block:: python
+
+        from functools import partial
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @partial(qml.transforms.split_non_commuting, grouping_strategy="wires")
+        @qml.qnode(dev)
+        def circuit(params):
+            qml.RX(params[0], wires=0)
+            qml.RZ(params[1], wires=1)
+            return [
+                qml.expval(qml.X(0)),
+                qml.expval(qml.Y(1)),
+                qml.expval(qml.Z(0) @ qml.Z(1)),
+                qml.expval(qml.X(0) @ qml.Z(1) + 0.5 * qml.Y(1) + qml.Z(0)),
+            ]
+
```

Comment:
```suggestion

Internally, the QNode is split into multiple circuits when executed:

>>> print(qml.draw(circuit)([np.pi/4, np.pi/4]))
0: ──RX(0.79)─┤  <X>
1: ──RZ(0.79)─┤  <Y>
<BLANKLINE>
0: ──RX(0.79)─┤ ╭<Z@Z>
1: ──RZ(0.79)─┤ ╰<Z@Z>
<BLANKLINE>
0: ──RX(0.79)─┤ ╭<X@Z>
1: ──RZ(0.79)─┤ ╰<X@Z>
<BLANKLINE>
0: ──RX(0.79)─┤  <Z>
1: ──RZ(0.79)─┤     
```

Reply:
Adding a line on how to run the circuit and what to expect as an output helps people debug.

Reply:
I actually don't even know if this result is correct, I'm assuming it is but I'm not even sure why we're not measuring qubit 1 in the last tape.

Reply:
I totally agree @CatalinaAlbornoz. Just in this case, I would personally avoid providing a line to test the circuit for 2 reasons:

- To keep consistency with the remaining examples in this file. For some reason, the code snippets shown in this file don't have an option to run the circuit, so I think it would be strange to insert one only here (OCD)

- This is basically the same example shown in the documentation for `split_non_commuting`, so users can find all the information they need if they access the page